### PR TITLE
[CORE-1818] Add documentation for User Group API tags

### PR DESCRIPTION
**Pull request by mbernat**
This PR documents User Group Tags and the API that's used to view and manage them.

One of the commits also fixes an incorrect example JSON for User Group Contact Details; it also adds missing `company_name` field to contact detail definition and examples.

---
This change is [Reviewable](https://reviewable.io/reviews/scrive/scrive-apidocs/78)

### CODE REVIEW

File: documentation/api/usergroup/updateusergroup.yaml [hunk 1]
```diff
@@ -37,6 +45,13 @@ post:
     as you have permissions to update the group being moved, the new parent and
     the old parent.
 
+    Tags are updated as follows:
+
+    + Only the provided tags are affected on the server.
+    + If the value of a tag is a string, the key/value pair is stored,
```
[the remark this answers is not in the document]
Comment (mbernat) [reply documentation/api/usergroup/updateusergroup.yaml]: You get a 400. Should this be mentioned in the API?

File: documentation/api/usergroup/getusergroup.yaml [hunk 2]
```diff
@@ -102,13 +112,22 @@ get:
           "inherited_from": "1",
           "address": {
             "company_number": "5568166804",
+            "company_name": "Scrive",
```
[the remark this answers is not in the document]
Comment (mbernat) [reply documentation/api/usergroup/getusergroup.yaml]: No problem, happy to help. Although these things will continue to go out of sync, I wonder whether there isn't a better way of generating examples.

File: documentation/definitions/UserGroup.yaml [hunk 3]
```diff
@@ -62,3 +68,5 @@ properties:
     $ref: ./UserGroupSettings.yaml
   contact_details:
     $ref: ./UserGroupContactDetails.yaml
+  tags:
+    $ref: ./UserGroupTags.yaml
```
[the remark this answers is not in the document]
Comment (mbernat) [reply documentation/definitions/UserGroup.yaml]: Thanks

File: documentation/api/usergroup/updateusergroup.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/usergroup/updateusergroup.yaml]: Yeah, add a bulletpoint for that case.

[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/usergroup/updateusergroup.yaml]: awesome. cheers

**Pull request merged**